### PR TITLE
Resets step reward buffer properly when weight is zero

### LINKED ISSUE #2391: [Bug Report] Step reward retains stale values when weight is dynamically set back to zero
### Describe the bug

In the reward computation logic (`compute` function), when a reward term has a weight of zero, the code skips computation and does not update `self._step_reward` for that term.  
This is not an issue if a term's weight remains zero from initialization or is changed from zero to non-zero after the reward manager is created.  
However, if a term's weight is first changed to non-zero and later back to zero, the `self._step_reward` would retain stale (nonzero) values from earlier updates, since the loop skips updating it when the weight is zero.

Explicitly setting `self._step_reward` to zero when the weight is zero ensures correctness and prevents stale data,

### Steps to reproduce

1. Create or initialize a reward manager with a reward term having zero weight.
2. Change the weight of the term dynamically to a non-zero value during runtime.
3. Compute the reward and observe that `self._step_reward` is correctly updated with nonzero values.
4. Change the term’s weight back to zero dynamically.
5. Compute the reward again.
6. Observe that `self._step_reward` still retains the previous nonzero value because the update is skipped when the weight is zero.

Example (pseudo-code):

```python
# Assume reward manager initialized normally
reward_manager._term_cfgs[idx].weight = 2.0  # Set to non-zero
reward_manager.compute(dt=0.02)

print(reward_manager._step_reward[:, idx])  # --> Should show nonzero value

reward_manager._term_cfgs[idx].weight = 0.0  # Set back to zero
reward_manager.compute(dt=0.02)

print(reward_manager._step_reward[:, idx])  
# Expected: 0.0
# Observed: still showing stale nonzero value
```

### System Info

Describe the characteristic of your environment:

- Commit: 2e6946afb9b26f6949d4b1fd0a00e9f4ef733fcc
- Isaac Lab Version: 2.1.0
- Isaac Sim Version: 4.5
- OS: Ubuntu 22.04
- GPU: RTX 3060
- CUDA: 12.4
- GPU Driver: 550.120

### Additional context

This bug was detected during live visualization of rewards in the IsaacSim GUI, where a curriculum reward setting was used.  
The `ManagerLiveVisualizer` relies on `_step_reward` to report reward terms visually, and in this case, it incorrectly reported stale reward values when the weight was dynamically changed back to zero.  
This issue affects any tool that uses `_step_reward` for visualization or logging.  

Note that the total reward (`_reward_buf`) computation remains correct, since skipping a term with zero weight or explicitly adding zero yields the same result.

### Checklist

- [x] I have checked that there is no similar issue in the repo (**required**)
- [x] I have checked that the issue is not in running Isaac Sim itself and is related to the repo

### Acceptance Criteria

Add the criteria for which this task is considered **done**. If not known at issue creation time, you can add this once the issue is assigned.

- [x] `self._step_reward` is explicitly set to zero for zero-weight terms
- [x] Per-term step reward accurately reflects zero contribution when weight is zero

### PR DESCRIPTION
**Pull request by bikcrum**
# Description

This pull request fixes a bug where `_step_reward` could retain stale values when a reward term's weight was dynamically changed back to zero.  
Previously, when a reward term had zero weight, the computation skipped updating `_step_reward`, assuming that it would stay correct.  
However, if the weight was first changed from zero to nonzero and then back to zero during runtime (e.g., in curriculum settings), stale nonzero values could persist, causing incorrect reward visualizations or logging.

This change explicitly sets `reward_manager._step_reward` to zero when a reward term has zero weight, ensuring correctness regardless of dynamic weight changes.

Fixes #2391 

No new dependencies are introduced by this change.

## Type of change

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Screenshots

_Not applicable._

## Checklist

- [x] I have run the [`pre-commit` checks](https://pre-commit.com/) with `./isaaclab.sh --format`
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have updated the changelog and the corresponding version in the extension's `config/extension.toml` file
- [x] I have added my name to the `CONTRIBUTORS.md` or my name already exists there

### CODE REVIEW

File: source/isaaclab/isaaclab/managers/reward_manager.py [hunk 1]
```diff
@@ -141,8 +141,11 @@ def compute(self, dt: float) -> torch.Tensor:
         self._reward_buf[:] = 0.0
         # iterate over all the reward terms
         for name, term_cfg in zip(self._term_names, self._term_cfgs):
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/managers/reward_manager.py]: Noticed this line of code because of this MR. Why do we index term_names in the loop to get the index of the reward term. Finding an item in a list in such a loop is expensive. It is easily fixable by just doing an enumerate with the zip.

Could this be fixed?

Comment (bikcrum) [reply source/isaaclab/isaaclab/managers/reward_manager.py]: @Mayankm96 Thank you for pointing that out. I have fixed that and pushed a change.

Comment (Mayankm96) [reply source/isaaclab/isaaclab/managers/reward_manager.py]: Thank you for the promptness!

**Review by Mayankm96 — APPROVED** [review]

File: source/isaaclab/isaaclab/managers/reward_manager.py [hunk 2]
```diff
@@ -140,9 +140,10 @@ def compute(self, dt: float) -> torch.Tensor:
         # reset computation
         self._reward_buf[:] = 0.0
         # iterate over all the reward terms
-        for name, term_cfg in zip(self._term_names, self._term_cfgs):
+        for term_idx, (name, term_cfg) in enumerate(zip(self._term_names, self._term_cfgs)):
             # skip if weight is zero (kind of a micro-optimization)
             if term_cfg.weight == 0.0:
+                self._step_reward[:, term_idx] = 0.0
```
Comment (Mayankm96) [inline source/isaaclab/isaaclab/managers/reward_manager.py:146]: Minor comment for optimization.

The weight is typically only changed through curriculum on resets, right? I wonder if the step reward should be made zero on resets only and not here between episodic steps. This way we can save an unnecessary operation while computing the rewards.

Comment (Mayankm96) [reply source/isaaclab/isaaclab/managers/reward_manager.py:146]: Though if you don't see any slowdown then I suppose we can keep it like this too.

Comment (bikcrum) [reply source/isaaclab/isaaclab/managers/reward_manager.py:146]: @Mayankm96 I agree that if reward weight is changed through the curriculum term, it will only be invoked during reset, which is true in most cases.

However, I have a special case where certain rewards need to be enabled or disabled dynamically at every step based on the values of other rewards. To support this, I use a hierarchical gated reward mechanism: parent rewards are monitored with threshold conditions, and once those are met, child rewards are activated. This check is performed in an event term rather than a curriculum term.

This setup is specific to my use case and likely not relevant to most users. 

That said, I ran a quick benchmark on my local RTX 3060. While `.zero_()` may provide a slight performance edge, the difference is negligible over time.

If my estimates are correct, with 8 zero-weight reward terms, 24 steps per iteration, and 1,000 iterations total, the additional cost of the current change amounts to roughly 1 extra second.

```python
import torch
```
```python
t = torch.randn(4096, 8).cuda()
```
```python
%%timeit
t[:, 0] = 0
```
> 6.2 μs ± 23.3 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)
```python
%%timeit
t[:, 0].zero_()
```
> 5.82 μs ± 22.9 ns per loop (mean ± std. dev. of 7 runs, 100,000 loops each)

#### Extra second calculation: 
6.2 μs per reward term x 8 reward terms per step x 24 steps per iteration x 1000 iterations
```python
(6.2 * 1e-6) * 8 * 24 * 1_000
```
> 1.1904

The computational overhead is minimal, and while moving the logic to the reset phase is possible, it would require users to manually update the private `_step_reward` buffer mid-episode if reward weights change, which may not be a clean workflow. In my opinion, keeping this logic in the step loop is a reasonable choice for correctness and abstraction.

Comment (Mayankm96) [reply source/isaaclab/isaaclab/managers/reward_manager.py:146]: Makes sense. Did not think of that use-case before :)

Thank you for the clarifications!
